### PR TITLE
Concurrency support in reaction graphs

### PR DESCRIPTION
**Pull request by sandhu**

### CODE REVIEW

**Comment by sandhu** [comment]
For review only. Not ready to merge

File: src/signum/signal.clj [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                     (>!! value-ch [new-value])))))
+    (let [stop-ch (chan)
+          value-ch (chan (sliding-buffer 1))]
+      (swap! watches assoc watch-key
+             {:stop-ch stop-ch
+              :value-ch value-ch
+              :value-loop (go-loop [old-value @backend]
+                            (let [[[value] channel] (alts! [stop-ch value-ch])]
+                              (when (not= channel stop-ch)
+                                (watch-fn watch-key this old-value value)
```
Comment (tom7t) [inline src/signum/signal.clj:44]: Is there any chance that watch-fn might throw? The go block will swallow the exception.

File: src/signum/signal.clj [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                                (recur value))))}))
+    this)
+  (removeWatch
+    [this watch-key]
+    (close! (get-in @watches [watch-key :stop-ch]))
```
Comment (tom7t) [inline src/signum/signal.clj:49]: Should value-ch be closed either here, or in the go-loop above in the event that stop-ch is closed?

File: src/signum/signal.clj (hunk 1 above)
Comment (sandhu) [reply src/signum/signal.clj:44]: Good catch. If the watch function throws we should dispose the subscription. Need to figure out how to notify the subscriber that we punted them.

File: src/signum/signal.clj (hunk 2 above)
Comment (sandhu) [reply src/signum/signal.clj:49]: Good catch.

File: src/signum/signal.clj (hunk 1 above)
Comment (tom7t) [reply src/signum/signal.clj:44]: Is this maybe where we would look into something like `https://github.com/IGJoshua/farolero` ?

Comment (tom7t) [reply src/signum/signal.clj:44]: Actually never mind, it requires a thread context and we're in core-async land.

**Pull request merged**